ROADMAP.md

```diff
@@ -0,0 +1,40 @@
+Roadmap
+=======
+
+These are the themes that we plan to work on for cert-manager. If you wish
+to discuss these topics you can find us in #cert-manager on Kubernetes Slack, or
+at our [community meetings](https://cert-manager.io/docs/contributing/#meetings).
+
+The roadmap items are categorized in to themes based on the larger goals we
+want to achieve with cert-manager.
+
+* Beyond Ingress: improve experience of cert-manager for applications beyond just
+  ingress certificates
+  * Service Mesh Integration: Enable service meshes to issue mTLS certificates
+    with cert-manager, getting the integration with external issuers and the
+    audit capabilities of cert-manager in their mesh
+  * Istio agent certificates issued via cert-manager
+  * CSI driver: seamlessly deliver unique certs + keys to workloads
+* Adoption of upstream APIs: continue to support latest APIs for k8s upstream
+  * CSR API: support CSR API as a standard for certificate requests in kubernetes
+* Policy: giving cluster admin granular control over certificate issuance
+  * Extensible primitives within cert-manager for defining policy for
```

Comment:
Worth thinking about how this would play with ecosystem projects like OPA

Should be extensible, eg a gRPC service that will be able to enforce it that can be ours or a 3rd party one. 
Or s similar system to DNS webhooks? 

How does this tie in with CSR API aproval

(brainstorm session from meeting)

Reply:
IMO, policy should be enforced during the approval flow as you noted here. In terms of how that policy is actually evaluated and applied, I don't think that needs to be a core concern of cert-manager itself as the 'plugin' mechanism already exists in that anyone can write their own approver.

If someone wants to use OPA to evaluate policy decisions, or call out to a gRPC web-service, that'd work just fine. But I think the purpose of having `approval` as part of the API already serves the purpose of making it extensible. I don't think cert-manager core should be opinionated about gRPC vs OPA etc

Reply:
I agree.

The fact that approval isn't something that we can easily do with a CRD today complicates this a little, but I think the goal of having the core be agnostic should still hold.